// Copyright (c) 2023 dingodb.com, Inc. All Rights Reserved
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "common.proto";
import "meta.proto";

package dingodb.pb.coordinator_internal;

// PartInternal is storage structure for coordinator
// PartInternal is not table partition in SQL, its only use for storage.
message PartInternal {
  uint64 region_id = 1;               // the region id of this part
  dingodb.pb.common.Range range = 2;  // the range of this part
}

// TableInternal is storage structure for coordinator
message TableInternal {
  uint64 id = 1;
  dingodb.pb.meta.TableDefinition definition = 2;
  repeated PartInternal partitions = 3;
}
